src/middleware.ts

```diff
@@ -61,3 +62,15 @@ export const verifyMiddleware = (excludes?: string[]) => (req: AuthRequest, res:
 export const errorMiddleware = (error: Error, req: Request, res: ApiResponse, next: NextFunction) => {
   return res.status(500).send({ message: `${error.name} ${error.message}`, result: null });
 };
+
+export const validateOrderData = (req: Request, res: ApiResponse, next: NextFunction) => {
+  try {
+    const orderData = req.body;
+
+    orderSchema.validateSync(orderData);
+
```

Comment:
middleware will be app-level mechanism 
move to controller level would be great